internal/plugin/convert/schema.go

```diff
@@ -43,6 +50,15 @@ func ConfigSchemaToProto(b *configschema.Block) *proto.Schema_Block {
 	return block
 }
 
+func protoStringKind(k configschema.StringKind) proto.StringKind {
+	switch k {
+	default:
+		return proto.StringKind_PLAIN
```

Comment:
nit: let's log an error here if this was not `configschema.StringPlain`, to ease debugging.